cmd/aws-k8s-agent/main.go

```diff
@@ -39,17 +39,21 @@ func _main() int {
 	version.RegisterMetric()
 
 	//Check API Server Connectivity
-	if k8sapi.CheckAPIServerConnectivity() != nil {
+	err := k8sapi.CheckAPIServerConnectivity()
```

Comment:
Doing a direct comparison to `nil` since there is no other return value seems better.

So, this line can reduced to simply adding a `log.Errorf`

Reply:
Do you mean this ?

```patch
--- a/cmd/aws-k8s-agent/main.go
+++ b/cmd/aws-k8s-agent/main.go
@@ -40,6 +40,7 @@ func _main() int {
 
        //Check API Server Connectivity
        if k8sapi.CheckAPIServerConnectivity() != nil {
+               log.Errorf("Failed to connect to the API server")
                return 1
        }
```
It would prevent to display the returned error while it contains valuable information.